bpf/lib/common.h

```diff
@@ -118,6 +119,17 @@ static __always_inline bool validate_ethertype(struct __ctx_buff *ctx,
 	void *data_end = ctx_data_end(ctx);
 	struct ethhdr *eth = data;
 
+#if __ctx_is == __ctx_skb
+	/* XDP does not support packets without L2 hdr, so skip the check.*/
```

Comment:
Given it's not possible to attach to XDP for those devices, why do we need the extra `__ctx_is == __ctx_skb` check?

Reply:
I wanted to compile out the case for XDP to save a few CPU cycles. Let me check whether compiler does it for us.

Reply:
OK, just checked - this one gets compiled away :+1: